rollup.config.js

```diff
@@ -83,6 +83,28 @@ function createConfigInternal({ format, production }) {
                     toplevel: format === 'cjs',
                     warnings: true,
                 }),
+            format === "cjs" && {
+                name: "writePkgJSON",
+                writeBundle: async () => {
+                    await fs.writeFile(
+                        "lib/cjs/package.json",
+                        JSON.stringify({
+                            type: "commonjs",
+                        })
+                    );
+                }
+            },
+            format === "es" && {
+                name: "writePkgJSON",
+                writeBundle: async () => {
+                    await fs.writeFile(
+                        "lib/package.json",
+                        JSON.stringify({
+                            type: "module",
+                        })
+                    );
+                }
+            },
```

Comment:
A `lib/package.json` with the contents `{"type":"module"}` and also a `lib/cjs/package.json` with the contents `{"type":"commonjs"}` is generated.

This tells Node.js to treat the contents of that folder/subfolder as the given file type without having to specify the `.cjs` or `.mjs` extension. I did not specify it in the root package.json as it caused issues with babel and jest (jest is not es ready yet).

Reply:
I also noticed that the lib folder also includes the "raw" js files besides the bundles. Do we really need those raw files? Why would one want to import those directly instead of using the top-level exports? Would it be enough to simply have the typescript definition files and the bundles?